frontend/src/pages/Delete.js

```diff
@@ -0,0 +1,36 @@
+import Header from "../components/Header";
+import Page from "../components/Page";
+import Main from "../components/Main";
+import {useAuth} from "../auth/AuthProvider";
+import {Link, Redirect} from "react-router-dom";
+import Button from "../components/Button";
+import {useState} from "react";
+
+
+export default function Delete() {
+    const {user, logout, deleteUser} = useAuth()
+    const [deletedUser, setDeletedUser] = useState()
+
+    const handleDelete = () => {
+        deleteUser(user.userName)
+            .then(deletedUser => setDeletedUser(deletedUser))
+            .then(logout)
+    }
+
+
+    if(deletedUser)
+    {
+        return <Redirect to = "/"/>
+    }
```

Comment:
```suggestion
    if(!user)
    {
        return <Redirect to = "/"/>
    }
```

So kommt auch ein nicht eingeloggter User nicht auf die Seite und durch das löschen des Token wird der User falsy und startet somit auch den Redirect.